openmc/tallies.py

```diff
@@ -2914,6 +3158,157 @@ def summation(self, scores=[], filter_type=None,
+            'cell', 'cellborn', and 'universe' Filters. Each bin is an integer
+            for the cell instance ID for 'distribcell' Filters. Each bin is a
+            2-tuple of floats for 'energy' and 'energyout' filters corresponding
+            to the energy boundaries of the bin of interest. Each bin is an
+            (x,y,z) 3-tuple for 'mesh' filters corresponding to the mesh cell of
+            interest.
+        nuclides : list of str
+            A list of nuclide name strings to average across
+            (e.g., ['U-235', 'U-238']; default is [])
+        remove_filter : bool
+            If a filter is being averaged over, this bool indicates whether to
+            remove that filter in the returned tally. Default is False.
+
+        Returns
+        -------
+        Tally
+            A new tally which encapsulates the average of data requested.
+        """
+
+        # Create new derived Tally for average
+        tally_avg = Tally()
+        tally_avg._derived = True
+        tally_avg._estimator = self.estimator
+        tally_avg._num_realizations = self.num_realizations
+        tally_avg._with_batch_statistics = self.with_batch_statistics
+        tally_avg._with_summary = self.with_summary
+        tally_avg._sp_filename = self._sp_filename
+        tally_avg._results_read = self._results_read
+
+        # Get tally data arrays reshaped with one dimension per filter
+        mean = self.get_reshaped_data(value='mean')
+        std_dev = self.get_reshaped_data(value='std_dev')
+
+        # Average across any filter bins specified by the user
+        if filter_type in _FILTER_TYPES:
+            find_filter = self.find_filter(filter_type)
+
+            # If user did not specify filter bins, average across all bins
+            if len(filter_bins) == 0:
+                bin_indices = np.arange(find_filter.num_bins)
+
+                if filter_type == 'distribcell':
+                    filter_bins = np.arange(find_filter.num_bins)
+                else:
+                    num_bins = find_filter.num_bins
+                    filter_bins = \
+                        [(find_filter.get_bin(i)) for i in range(num_bins)]
+
+            # Only average across bins specified by the user
+            else:
+                bin_indices = \
+                    [find_filter.get_bin_index(bin) for bin in filter_bins]
+
+            # Average across the bins in the user-specified filter
+            for i, self_filter in enumerate(self.filters):
+                if self_filter.type == filter_type:
+                    mean = np.take(mean, indices=bin_indices, axis=i)
+                    std_dev = np.take(std_dev, indices=bin_indices, axis=i)
+                    mean = np.mean(mean, axis=i, keepdims=True)
+                    std_dev = np.mean(std_dev**2, axis=i, keepdims=True)
+                    std_dev /= len(bin_indices)
+                    std_dev = np.sqrt(std_dev)
```

Comment:
I don't think so - my implementation here reflects my understanding of the equation for the standard deviation in the table of "Example Formulas" in cell (3,2) on this [wiki](https://en.wikipedia.org/wiki/Propagation_of_uncertainty) page. The first line calls `np.mean(...)` of the squared standard deviation (variance) divided by n, the second line applies the second (1/n) factor to get the squared coefficient in the equation, and the final line takes the square root of the sum to back out the final standard deviation. If I